File: src/PatikaBeosztas.Domain/WorkPreference.cs
namespace PatikaBeosztas.Domain;

public sealed class WorkPreference
{
    public Guid Id { get; set; }

    public Guid OrganizationId { get; set; }

    public Guid EmployeeId { get; set; }

    public WorkPreferenceType Type { get; set; }

    public DateOnly DateFrom { get; set; }

    public DateOnly DateTo { get; set; }

    public DayOfWeek? DayOfWeek { get; set; }

    public bool IsFullDay { get; set; }

    public TimeOnly? StartTime { get; set; }

    public TimeOnly? EndTime { get; set; }

    public Guid? LocationId { get; set; }

    public string? Note { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTimeOffset CreatedAtUtc { get; set; }

    public DateTimeOffset UpdatedAtUtc { get; set; }

    public uint Version { get; private set; }

    public Employee? Employee { get; set; }

    public Location? Location { get; set; }
}

File: frontend/src/services/http/dto/work-preference.ts
// Munkavégzési kérések és visszatérő szabályok DTO-i.
// Backend szerződés: contracts/openapi.phase2d.json (Work preferences tag).

import type { BackendDayOfWeek } from "./enums";

export type BackendWorkPreferenceType =
  "Available" | "Preferred" | "Avoid" | "Unavailable" | "Fixed";

export interface WorkPreferenceResponseDto {
  id: string;
  employeeId: string;
  employeeDisplayName: string;
  type: BackendWorkPreferenceType;
  dateFrom: string;
  dateTo: string;
  dayOfWeek: BackendDayOfWeek | null;
  isFullDay: boolean;
  startTime: string | null;
  endTime: string | null;
  locationId: string | null;
  locationName: string | null;
  note: string | null;
  isActive: boolean;
  version: number | string;
  createdAtUtc: string;
  updatedAtUtc: string;
}

export interface CreateWorkPreferenceRequestDto {
  type: BackendWorkPreferenceType;
  dateFrom: string;
  dateTo: string;
  dayOfWeek: BackendDayOfWeek | null;
  isFullDay: boolean;
  startTime: string | null;
  endTime: string | null;
  locationId: string | null;
  note: string | null;
}

export interface UpdateWorkPreferenceRequestDto extends CreateWorkPreferenceRequestDto {
  expectedVersion: number;
}

export interface DeactivateWorkPreferenceRequestDto {
  expectedVersion: number;
}
